python/lsst/dax/webservcommon/auth.py

```diff
@@ -0,0 +1,72 @@
+import json
+from functools import wraps
+
+import jwt
+import requests
+from flask import request, g
+from jwt.algorithms import RSAAlgorithm
+
+ALGORITHM = "RS256"
+
+
+class DaxAuthException(Exception):
+    pass
+
+
+class JwtAuth:
+
+    def __init__(self, key, issuer=None, audience=None, options=None):
+        """
+        JwtAuth is a thin wrapper around the PyJWT project. Parameters
+        are best documented in that project.
+        :param key: Public key to verify RS256 signatures with
+        :param issuer: Domain of issuer, if applicable
+        :param audience: Audience of token, if applicable. If this is
+        false, then the check is turned off
+        :param options: verification options to pass to decode
+        """
+        if key is None:
+            raise DaxAuthException("Need a valid public key for initialization")
+        self.key = key
+        self.issuer = issuer
+        self.options = options or {}
+        self.audience = audience
+        if not self.audience and "verify_aud" not in self.options:
+            self.options["verify_aud"] = False
+
+    def jwt_required(self):
+        """View decorator that requires a valid JWT token and adds"""
+
+        def wrapper(fn):
+            @wraps(fn)
+            def decorator(*args, **kwargs):
+                self._validate()
+                return fn(*args, **kwargs)
+
+            return decorator
+
+        return wrapper
+
+    def _validate(self):
```

Comment:
Some documentation would be good even for private method.

Reply:
done